cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataHistoryEntry.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata;
+
+import co.cask.cdap.proto.Id;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents the complete metadata of a {@link Id.NamespacedId} including its properties and tags.
+ */
+public class MetadataHistoryEntry {
+  private final Id.NamespacedId entityId;
+  private final Map<String, String> properties;
+  private final Set<String> tags;
+
+  /**
+   * Returns an empty {@link MetadataHistoryEntry}
+   */
+  public MetadataHistoryEntry(Id.NamespacedId entityId) {
```

Comment:
we shouldn't need these variants either right? As far as possible, lets only have the constructor on line 46